benchmarks/communication/all_gather.py

```diff
@@ -16,7 +16,10 @@
 
 
 # Run all_gather and print metrics
-def timed_all_gather(input, output, start_event, end_event, args):
+def timed_all_gather(input, output, start_event, end_event, args, device):
```

Comment:
Why the change in interface to explicitly send the device? Is it not part of the `args` parameter (e.g. `args.device`)? Are there cases where `device != args.device`?